### ISSUE TITLE
Result Source ignored

### ISSUE THREAD

**Issue by outorted** [microsoft-search/pnp-modern-search#1918]
Hi,

a few weeks ago, everything was working fine. Now, our PnP Search Results webparts are ignoring the result sources and showing results from everywhere in SharePoint, including the current user's OneDrive, Teams, and other SharePoint sites (in the same geo and also other geo locations). We were using Modern Search v3, but for troubleshooting purposes, I rebuilt everything with v4 - and got the same results.

To reproduce the issue, create a result source using the OpenSearch protocol, {searchTerms} as Query Transform and enter the source Url. Ours was in the format: 
https://(website)/de-de/api-search/documentation?term={searchTerms}&lang=DE,EN&source=ABC and uses anonymous authentication. If you need the exact URL, please let me know and I can provide it (although it can be accessed anonymously, I didn't want to post it here in the forum for everyone to see). If I enter this URL directly into the browser and replace {searchterms} with the relevant search term, I get 61 results. 

I then created a site page with a PnP search webpart and a PnP search results webpart. I connected the search results webpart to the result source using its GUID, and also to the search webpart. I did this with v4 of the PnP Search webparts. I am then getting 326 results. None of them are coming from the external result source; they are all coming from somewhere within M365.

I updated to the latest versions of v3 and v4 yesterday and that didn't help.

Do you have any idea what is going on here and how to fix the issue?

Thanks

**Comment by outorted** [comment]
I've just realised it's also the same for result sources that query local SharePoint. The format is the following:
path:https://(SiteUrl)/Lists/Marketing/ (contentclass:STS_ListItem_DocumentLibrary) owstaxIdField1:{searchTerms} 

This is how I'm contacting the Result Source
![image](https://user-images.githubusercontent.com/94896540/157389180-a7cb9660-af05-4574-9084-02b18c3aabb8.png)

...and I'm getting that GUID from the URL of the result source:
![image](https://user-images.githubusercontent.com/94896540/157389430-0e93baf1-1e10-49c1-ab67-bfce72864a0d.png)

If I manually change the GUID in the search webpart configuration, nothing changes (I carry on getting hundreds of results but not the ones I want). As long as the length of the GUID is correct, there is no error message. But I get the same results with a fake GUID as with a real one. I've tried putting it in {braces} but that didn't help.

**Comment by patrikhellgren** [comment]
@outorted We have also been seeing the same thing with OpenSearch result sources on and off for a couple of months. For us it has been only with newly created sources, older ones has still been working. We opened a case with MS Support and got it resolved a few weeks ago (at least in the tenants where we saw the issue). However two weeks ago we started seeing this again and opened another case which got resolved and closed february 28 with the comment that a "global fix" had been deployed by the product group.
This has as you can see nothing to do with the Modern Search web parts and you can probably reproduce the issue using e.g. Search Query Tool.
I would also recommend that you open a case with Microsoft Support regarding this so that they are made aware of that the issue is still remaining in some form.

**Comment by outorted** [comment]
Hi @patrikhellgren thanks for the info. Have you been noticing issues with SharePoint result sources as well or just with OpenSearch? I added a comment just after my original post, when I noticed that our SharePoint result sources were also getting ignored.

**Comment by patrikhellgren** [comment]
@outorted I didn't have any SharePoint result sources to test with so I created a new one and that seems to be working ok. But as I mentioned we have seen differences in behaviour between old result sources and newly created ones so this might not have been a valid test.

**Comment by outorted** [comment]
Hi @patrikhellgren,

I've got a couple of steps further... First off, I downloaded the SharePoint Search Query Tool (great tool, by the way :-)). That was working fine and displayed the correct results for both my SharePoint searches and my OpenSearch searches.

After a bit of trial-and-error, I found a workaround for OpenSearch (it works in both the v3 and v4 webpart), but I haven't got it to work for SharePoint results.
The workaround for OpenSearch was to add the SharePoint site path (the site that I use to initiate the search) as a Refinement Filter:
SitePath:equals("https://tenantUrl.sharepoint.com/sites/sitename"). It then displayed the correct results from the external source, instead of everything that it could find in SharePoint. The OpenSearch is pointing to a completely different website (not SharePoint), so I'm not quite sure why this works, but it does :-).

With a result source pointing to a SharePoint library on the same site, however, I still can't get it to work. If I don't provide any refinement filters, the webpart shows hundreds of results from all over SharePoint/OneDrive/Teams. If I add the refinement filter, then it displays 0 results. If I do the same in the Search Query Tool, it shows the correct results if I don't define any refinement filters. If I add the refinement filter, it returns 0 results. I tried experimenting with some other possible refinement filters, but couldn't find anything that returned anything other than 0 results.

So if the Query Tool is displaying the correct results, then I think the issue might not be on Microsoft's side, after all. 

If it helps, this is the Query Transform on the Result Source for Local SharePoint:
path:tenantUrl/sites/sitename/Lists/LibraryName/ (contentclass:STS_ListItem_DocumentLibrary) owstaxIdManagedMetadataField1:{searchTerms}  

If you have any other ideas, I'd appreciate it.

Thanks
